modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/connectivity/ExtensionsConnectionAdapter.java

```diff
@@ -56,13 +54,36 @@
    * @throws ConnectionException  if connection could not be obtained
    * @throws TransactionException if something is wrong with the transaction
    */
-  public ConnectionHandler<?> getConnection(ExecutionContextAdapter<ComponentModel> executionContext)
+  public ConnectionHandler<?> getConnection(ExecutionContextAdapter<? extends ComponentModel> executionContext)
       throws ConnectionException, TransactionException {
     return executionContext.getTransactionConfig().isPresent()
         ? getTransactedConnectionHandler(executionContext, executionContext.getTransactionConfig().get())
         : getTransactionlessConnectionHandler(executionContext);
   }
 
+  private <T extends TransactionalConnection> ConnectionHandler<T> getTransactedConnectionHandler(
+                                                                                                  ExecutionContextAdapter<? extends ComponentModel> executionContext,
+                                                                                                  TransactionConfig transactionConfig)
+      throws ConnectionException, TransactionException {
+
+    if (transactionConfig.getAction() == ACTION_NOT_SUPPORTED || transactionConfig.getAction() == ACTION_NONE) {
```

Comment:
it's not a lie. It's a very bad case of naming which we're keeping because of trying to playing nice with JTA. Not steve's fault and a decision was made to roll with this